agent/php_execute.c

```diff
@@ -2244,39 +1982,28 @@ static void nr_php_instrument_func_end(NR_EXECUTE_PROTO) {
      */
     return;
   }
-  if (nrunlikely(NULL == segment->metadata)) {
-    /*
-     * If this value isn't set, it is either the root segment not a stacked
-     * segment set or not set by the instrument_begin_func, but in all we we
-     * should only ignore it.
-     */
+  if (nrunlikely(NRPRG(txn)->segment_root == segment)) {
     return;
   }
-  /*
-   * If the metadata info doesn't match, an uncaught exception occurred which
-   * doesn't call fcall_end.
-   */
-  metadata = segment->metadata;
-  if ((metadata->execute_data_this != &execute_data->This)) {
+
+  wraprec = segment->wraprec;
+
+  if (wraprec && wraprec->is_exception_handler) {
```

Comment:
This `if` statement doesn't seem to contain anything anymore and should be removed

Reply:
I added clarifying comments in [bce3e14](https://github.com/newrelic/newrelic-php-agent/pull/767/commits/bce3e1462db26bb5f841542c199dae92ba96a1b3). I had this split off as an empty if statement (as opposed to combining this check with the `else if` check) to explicitly call out each case and have an explaining comment with each case. Do you still want the if's combined?

Reply:
Having a hollow conditional doesn't quite sit right with me. We don't have to combine the `if`'s necessarily if that's something you want to avoid, we could invert the logic and have the else if/else statements be nested within, right? Something like: 
```
if (!wraprec || !wraprec->is_exception_handler) {...}
```
Then just nest the `else if` statement as an `if` within it?

Reply:
see https://github.com/newrelic/newrelic-php-agent/pull/825